From f974e0e6eca0057c456f73474d878127b827835e Mon Sep 17 00:00:00 2001
From: Chad Dupuis <cdupuis@redhat.com>
Date: Thu, 21 Aug 2014 13:05:29 -0400
Subject: [ethernet] qlcnic: Fix TSS/RSS validation for 83xx/84xx series adapter

Message-id: <1408626422-3941-46-git-send-email-cdupuis@redhat.com>
Patchwork-id: 88577
O-Subject: [PATCH RHEL 7.1 045/138] qlcnic: Fix TSS/RSS validation for 83xx/84xx series adapter.
Bugzilla: 1089352
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: Himanshu Madhani <himanshu.madhani@qlogic.com>

Bug 1089352 (https://bugzilla.redhat.com/show_bug.cgi?id=1089352)

net commit id b17a44d8b86f48e34011b884a934231ae2928d66

o Current code was not allowing the user to configure more
  than one Tx ring using ethtool for 83xx/84xx adapter.
  This regression was introduced by commit id
  18afc102fdcb95d6c7d57f2967a06f2f8fe3ba4c ("qlcnic: Enable
  multiple Tx queue support for 83xx/84xx Series adapter.")

Signed-off-by: Himanshu Madhani <himanshu.madhani@qlogic.com>
Signed-off-by: David S. Miller <davem@davemloft.net>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
index 78f5e81..e3be276 100644
--- a/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
+++ b/drivers/net/ethernet/qlogic/qlcnic/qlcnic_ethtool.c
@@ -680,21 +680,12 @@ static int qlcnic_validate_ring_count(struct qlcnic_adapter *adapter,
  }
 
   if (tx_ring != 0) {
-  if (qlcnic_82xx_check(adapter) &&
-      (tx_ring > adapter->max_tx_rings)) {
+  if (tx_ring > adapter->max_tx_rings) {
    netdev_err(adapter->netdev,
        "Invalid ring count, Tx ring count %d should not be greater than max %d driver Tx rings.\n",
        tx_ring, adapter->max_tx_rings);
    return -EINVAL;
   }
-
-  if (qlcnic_83xx_check(adapter) &&
-      (tx_ring > QLCNIC_SINGLE_RING)) {
-   netdev_err(adapter->netdev,
-       "Invalid ring count, Tx ring count %d should not be greater than %d driver Tx rings.\n",
-       tx_ring, QLCNIC_SINGLE_RING);
-    return -EINVAL;
-  }
  }
 
  return 0;
-- 
1.7.1